screens/AddRespondersScreen.js

```diff
@@ -0,0 +1,146 @@
+import React, { Component } from "react";
+import { StyleSheet } from "react-native";
+import { Actions } from "react-native-router-flux";
+import { Container, Content, Header, View, List, ListItem } from "../components/layout";
+import { Checkbox, SearchBar, Input } from "../components/forms";
+import { Text } from "../components/typography";
+import { Button } from "../components/buttons";
+import theme from "../styles/base";
+
+// fake data
+const fakeResponders = [
+  "a",
+  "aa",
+  "ab",
+  "b",
+  "ba",
+  "bb",
+  "bc",
+  "c",
+  "ca",
+  "cb",
+  "cc",
+];
+
+class AddRespondersScreen extends Component {
+  constructor(props) {
+    super(props);
+    // TODO: fetch all responders (& put in alphabetical order)
+    let responderSelectionMap = new Map();
+    for (username of fakeResponders) {
+      responderSelectionMap.set(username, false);
+    }
+    this.state = {
+      searchQuery: "",
+      responderSelectionMap: responderSelectionMap,
+      allResponders: fakeResponders,
+      queriedResponders: fakeResponders
+    };
+  }
+
+  onCheckboxPress = (username) => {
+    let responderSelectionMap = this.state.responderSelectionMap;
+    let oldValue = responderSelectionMap.get(username);
+    responderSelectionMap.set(username, !oldValue);
+    this.setState({responderSelectionMap});
+  }
+
+  regexEscape = (str) => {
+    return str.replace(/[-\/\\^$*+?.()|[\]{}]/g, "\\$&");
+  }
+
+  searchForUsername = () => {
+    const searchQuery = this.regexEscape(this.state.searchQuery);
+    const regex = new RegExp("^" + searchQuery, "i");
+    let queriedResponders = [];
+    for (username of this.state.allResponders) {
+      if (regex.test(username)) {
+        queriedResponders.push(username);
+      }
+    }
+    this.setState({queriedResponders});
+  }
+
+  addSelectedResponders = () => {
+    let selectedResponders = [];
+    for (let [username, isSelected] of this.state.responderSelectionMap) {
+      if (isSelected) {
+        selectedResponders.push(username);
+      }
+    }
+    // TODO: ping backend to add the usernames in selectedResponders as responders for this user
+    
+    Actions.pop();
+  }
+
+  render() {
+    return (
+      <Container>
+        <Header leftButton="arrow" 
+        onLeftButtonPress={() => Actions.pop()}
+        >
+          Add Responders
+        </Header>
+
+        <Content>
+          <View style={styles.search}>
+            <SearchBar placeholder="Search for a username"
+              enableClearButton
+              onChangeText={(searchQuery) => this.setState({searchQuery})}
+              onClearButtonPress={() => this.setState({searchQuery: ""})}
+              onSubmitEditing={() => this.searchForUsername()} 
```

Comment:
Ideally I want the filtered results based on query to be shown when they are entering the query. Seems like onSubmitEditing will only do it when you "enter" the query (which is working fine with this code). Have you explored the option of auto-filtering when the query length > some arbitrary number?